primer/src/Primer/Eval.hs

```diff
@@ -130,6 +130,7 @@ evalOpts =
     { groupedLets = True
     , aggressiveElision = True
     , avoidShadowing = True
+    -- TODO: what option do we want here? NB neither is "more permissive", as it swaps "rename" and "push". If change this, will need to think about our test!
     }
```

Comment:
Bikeshed: are people happy with enforcing single-step eval to always avoid shadowing? I could factor this out into a non-hardcoded option if we want.

Reply:
Let's leave it as an option, if it's not *too* much extra work.

Reply:
It is not much work, it is already an option in the core of eval, just needs piping around through all the relevant functions rather than being picked up from this hardcoded set. (The hardcoded was initially "maximally permissive" for the interactive eval: the student should be able to choose any redex, but since has diverged from that: `aggressiveElision` is somewhat sensible to cut down on "bookkeeping"; `avoidShadowing` should arguably not even be an option, but allow two redexes on the same node "reduce and create shadow" and "rename to avoid later shadow", but we don't have that ability currently).

Reply:
The only other objection I'd have is if it slowed down eval, but I think that in single-step mode, it won't matter. This is assuming, of course, that we end up writing a separate fast evaluator for full/run-to-breakpoint evaluation.